docs/sources/k6/next/misc/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,265 @@
+> *It MAY be incremented if substantial new functionality or improvements are introduced within the private code.*
+> *It MAY include patch-level changes. The patch version MUST be reset to 0 when the minor version is incremented.*
+
+#### Patch version
+
+A patch version contains bug fixes **that do not affect** the API.
+
+Quoting the [semver 2.0.0 specification on patch versions](https://semver.org/#spec-item-6) directly:
+
+> *Patch version Z (x.y.Z | x \> 0\) **MUST** be incremented if only backward compatible bug fixes are introduced. A bug fix is defined as an internal change that fixes incorrect behavior.*
+### Breaking change
+
+As per the [Semantic Versioning 2.0.0](https://semver.org/spec/v2.0.0.html) specification, we define a breaking change as a change to k6 that introduces a change that is not backward compatible.
+
+Breaking changes imply that the consumer of k6 is expected to make some effort to adapt to the changes being made.
+
+Although it is possible that, in certain cases, the new major version containing breaking changes might be compatible with the previous version, this is not expected.
+
+## Release strategy
+
+A k6 release is a set of changes that updates or adds new functionalities to the product, made globally available under a unique and new identifiable version.
+
+k6 follows and aligns with the [Grafana Labs stages](https://grafana.com/docs/release-life-cycle). Not all changes follow a strict and sequential path through all the various steps. Most of the changes directly pass from Experimental to General Available.
+
+### Preview
+
+A preview release includes changes that should be considered unstable or under development but are ready for users to start testing changes or to prepare their migrations.
+
+As soon as the development team starts to work on a new set of changes or breaking changes, opening the need for a new version, a next version should be available under a dedicated versioning format suffixing the `-next` text.
+
+For example, if v1.3.5 is the latest major release, all the newly introduced breaking changes will be available under the `v2.0.0-next` version.
+Instead, all the backward-compatible changes will be available under `v1.4.0-next` or `v1.3.6-next`.
+
+### Frequency
+
+#### Major versions
+
+We aim to release a new major version once a year, in alignment with the practice observed in the Grafana ecosystem. This affords us to maintain a regular rhythm, offering opportunities for better feature planning, communication and visibility around releases, and regular cleanup of deprecated functionalities, helping reduce technical debt over time.
+
+#### Minor versions
+
+Minor versions are expected to be released every six weeks as a part of a regular process.
+
+#### Patch versions
+
+A new Patch version is released only if needed and is priority-based. Otherwise, fixes and changes that don’t introduce a new API or change the existing one will be released with the next minor version.
```

Comment:
```suggestion
A new patch version is released only if needed and is priority-based. Otherwise, fixes and changes that don’t introduce a new API or change the existing one will be released with the next minor version.
```